draft-ietf-babel-information-model.md

```diff
@@ -389,61 +516,89 @@ babel-interface-metric-algorithm:
 babel-mcast-hello-seqno:
 : The current sequence number in use for multicast
   hellos sent on this interface.
+  This is a 16-bit unsigned integer.
 
 babel-mcast-hello-interval:
 : The current interval in use for multicast hellos
   sent on this interface.
+  This is a 16-bit unsigned integer.
 
 babel-update-interval:
 : The current interval in use for all updates (multicast
   and unicast) sent on this interface.
+  This is a 16-bit unsigned integer.
 
 babel-message-log-enable:
-: When written, it configures whether logging should be enabled
-  (true) or disabled (false).
-  A read from the running or intended datastore indicates the
-  configured administrative value of whether logging is enabled
-  (true) or not (false). A read from the operational datastore indicates whether
-  logging is actually running (true) or not (i.e., it indicates the
-  operational state).
-  A data model that does not replicate parameters for running and operational
-  datastores can implement this as two separate parameters.
-  An implementation MAY choose
-  to expose this parameter as read-only ("ro").
+: Indicates whether message logging is enabled
+  (true) or disabled (false) on this interface.
 
 babel-message-log:
-: Log entries that have timestamp of a received Babel message
-  and the entire received Babel
-  message  (including Ethernet frame and IP headers,
-  if possible). An implementation must restrict the size of this log, but how
-  and what size is implementation-specific.
-  If this log is implemented, a mechanism
-  to clear it SHOULD be provided.
+: A reference or url link to a file that contains a timestamped log
+  of messages received and sent on babel-udp-port on this interface.
+  The {{libpcap}} file format with .pcap file extension SHOULD be supported for
+  message log files. Logging is
+  enabled / disabled by babel-message-log-enable.
+
+babel-stats-enable:
+: Indicates whether statistics collection is enabled
+  (true) or disabled (false) on this interface, including
+  neighbor-specific statistics (babel-nbr-stats).

```

Comment:
I might have commented this somewhere else, but I am not clear on why one would want to enable/disable stats. An implementation may choose to or not to implement stats, but if implemented and collected, one could choose not to read the stats that are being collected. Why force an implementation to or not to collect stats?

This also makes implementation of what the stats reflect, i.e. stats w.r.t. start of time, or w.r.t. to last clear, much more difficult. See below.

Reply:
Sometimes stats collection can cause performance issues. Some of the routers that run Babel are resource-constrained. We can take this to the list to see what others think.

Reply:
Ok. As I mentioned, if collecting stats can cause a performance issue, generally implementations will not even implement statistics gathering. Having a flag that enables/disables means I can enable stat collection and actually cause performance issues. But as you said, best to take it to the list, as you have already done.